src/main.nr

```diff
@@ -1,19 +1,68 @@
+use dep::std;
 global PRIME: u32 = 16777619;
 
+
 fn main(value: u32, hash_value: pub u32){
     let calculated_hash_value = calculate_hash(value);
-
+    
+    std::println(calculated_hash_value);
     assert(calculated_hash_value == hash_value);
 }
 
 fn calculate_hash(num: u32) -> u32 {
+
+    let size: Field = 2.pow_32(32);
+    let size_as_u32: u32 = size as u32;
```

Comment:
2**32 cannot fit into u32. If you print size_as_u32 and check you will see value "0x00". The application panic error is because of this as you are later doing mod with 0